recipes/hdf4/all/conanfile.py

```diff
@@ -0,0 +1,121 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class Hdf4Conan(ConanFile):
+    name = "hdf4"
+    description = "HDF4 is a data model, library, and file format for storing and managing data."
+    license = "BSD-3-Clause"
+    topics = ("conan", "hdf4", "hdf", "data")
+    homepage = "https://portal.hdfgroup.org/display/HDF4/HDF4"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "jpegturbo": [True, False],
+        "szip_support": ["None", "with_libaec", "with_szip"],
+        "szip_encoding": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "jpegturbo": False,
+        "szip_support": "None",
+        "szip_encoding": False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.szip_support == "None":
```

Comment:
```suggestion
        if self.options.szip_support is None:
```

Reply:
I think this is done for the same reason as this pr https://github.com/conan-io/conan-center-index/pull/1058

Reply:
@SpaceIm Can you check nonetheless that this recipe does not suffer from this weird behaviour of options?

Reply:
Indeed, it's the same issue, thanks !